library/LuaTypes.cpp

```diff
@@ -1422,6 +1422,28 @@ void struct_identity::build_metatable(lua_State *state)
     SetPtrMethods(state, base+1, base+2);
 }
 
+void other_vectors_identity::build_metatable(lua_State *state)
+{
+    int base = lua_gettop(state);
+    MakeFieldMetatable(state, this, meta_struct_index, meta_struct_newindex);
+
+    EnableMetaField(state, base+2, "_enum");
+
+    LookupInTable(state, index_enum, &DFHACK_TYPEID_TABLE_TOKEN);
+    lua_setfield(state, base+1, "_enum");
+
+    auto keys = &index_enum->getKeys()[-index_enum->getFirstItem()];
```

Comment:
I'm not sure I follow the logic here - why is this negated? Wouldn't this end up being `1` for types with `NONE = -1`?

Reply:
Ok, I think I see why - getKeys() starts with the first enum item (-1) at index 0, but you want keys starting at the second enum item (to skip NONE), which is index 1 in the array.
I guess this actually does work for enums starting at any non-positive value, although I think it could stand to be clarified a bit.

Reply:
yeah, we're taking the address of key 0

it assumes that the enum isn't a complex enum, so that should probably at least have an assert